internal/scriptrun/scriptrun.go

```diff
@@ -0,0 +1,197 @@
+package scriptrun
+
+import (
+	"os"
+	"path/filepath"
+	rt "runtime"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/failures"
+	"github.com/ActiveState/cli/internal/language"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/runbits"
+	"github.com/ActiveState/cli/internal/scriptfile"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/internal/virtualenvironment"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+var (
+	// FailScriptNotDefined indicates the user provided a script name that is not defined
+	FailScriptNotDefined = failures.Type("run.fail.scriptnotfound", failures.FailUser)
+	// FailStandaloneConflict indicates when a script is run standalone, but unable to be so
+	FailStandaloneConflict = failures.Type("run.fail.standaloneconflict", failures.FailUser)
+	// FailExecNotFound indicates when the builtin language exec is not available
+	FailExecNotFound = failures.Type("run.fail.execnotfound", failures.FailUser)
+)
+
+// ProjectHasScript is a helper function to determine if a project contains a
+// script by name.
+func ProjectHasScript(proj *project.Project, name string) bool {
+	script := proj.ScriptByName(name)
+	return script != nil
```

Comment:
This is pointless busywork. Please delete this function and just use `proj.ScriptByName(name) != nil`.

Reply:
This is needed by the run runner so that it only prints the title if it is somewhat reasonable that the script will run correctly.